src/components/va-back-to-top/va-back-to-top.tsx

```diff
@@ -3,6 +3,16 @@ import classnames from 'classnames';
 
 import 'intersection-observer';
 
+/**
+ * This component has three main behaviors:
+ * 1. Going to the top of the page when the button is clicked
+ * 2. "Revealing" after scrolling down far enough
+ * 3. "Docking" when the dock becomes visible
+ *
+ * The span.reveal-point determines when the reveal happens.
+ * It is positioned on the page in an absolute manner, and the
+ * button will reveal when the span is above the viewport.
```

Comment:
```suggestion
 * It uses `position: absolute` so the button will reveal when
 * the `span` is above the viewport.
```

This doc comment is awesome. Thanks!